src/core/schema.js

```diff
@@ -138,6 +138,7 @@ function parseProperty (value, propDefinition) {
   // Use default value if value is falsy.
   if (value === undefined || value === null || value === '') {
     value = propDefinition.default;
+    if (Array.isArray(value)) { value = value.slice(); }
```

Comment:
What do you mean by objects? The only types we have that can receive an object are `vec2`, `vec3` and `vec4` and those are already copied when parsed: https://github.com/aframevr/aframe/blob/master/src/utils/coordinates.js#L17